modules/nw-ingress-gateway-api-deployment-topologies.adoc

```diff
@@ -0,0 +1,135 @@
+// Modules included in the following assemblies:
+//
+// * networking/gateway-api.adoc
+
+:_mod-docs-content-type: CONCEPT
+[id="nw-ingress-gateway-api-deployment_{context}"]
+= Gateway API Deployment topologies
+
+Gateway API is designed to accomodate two topologies: shared gateways or dedicated gateways. Each topology has their own advantages and as well as different security implications.
+
+Dedicated gateway:: Routes and any load balancers or proxies are served from the same namespace. The `Gateway`
+object restricts routes to a particular application namespace. This is the default topology for Gateway API in {product-title}.
+
+Shared gateway:: Routes are served from multiple namespaces or with multiple hostnames. The `Gateway` object filters allow routes from application namespaces using the `spec.listeners.allowedRoutes.namespaces` field.
+
+[id="dedicated-gateway-example_{context}"]
+== Dedicated gateway example
+An example of a `Gateway` resource, `fin-gateway`:
+
+.Example dedicated `Gateway` resource
+[source,yaml]
+----
+apiVersion: gateway.networking.k8s.io/v1
+kind: Gateway
+metadata:
+  name: fin-gateway
+  namespace: openshift-ingress
+spec:
+  listeners:
+  - name: http
+    protocol: HTTP
+    port: 8080
+    hostname: "example.com"
```

Comment:
I think the dedicated gateway can also serve multiple httproutes in `Same` namespace, so could use wildcard hostname like `hostname: "*.example.com"`
@alebedev87 Do you think we should consider the DNSRecords creation in the example, if yes then need to set hostname like `*.dedicated.<baseDomain>`, the baseDomin could be get from `oc get dns.config cluster -ojsonpath="{.spec.baseDomain}"`

Reply:
this sounds like it can be an continuous improvement task for a new PR. the current one is pretty large and has had enough scope creep already